### PR TITLE
JSAE-37: Mock List products for backend, test covered

### PR DESCRIPTION
**Pull request by Jeffzholy**

### CODE REVIEW

File: back-end/routers/productsRoute.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const express = require("express");
+const router = express.Router();
+
+const { authenticate } = require("../services/productsService");
+const { generalResponse } = require("../helpers/responses");
+const { mockData, mockError } = require("../helpers/mock-data");
```
Comment (adamgyulavari) [inline back-end/routers/productsRoute.js]: @Jeffzholy make your life easier with `const mock = require(...);` so you can refer to it with `mock.data` and `mock.error`

File: back-end/routers/productsRoute.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+router.route("/").get((req, res) => {
+  const { headers } = req;
+  if (authenticate(headers)) {
+    return generalResponse(200, res, mockData);
```
Comment (adamgyulavari) [inline back-end/routers/productsRoute.js]: @Jeffzholy generally the better practice is to switch the 2 things: if not authenticated then do something, but the main part of the handler should be the normal valid response

Comment (adamgyulavari) [reply back-end/routers/productsRoute.js]: `generalResponse` is not going to return anything and right now it's not worth to hide the functionality behind a method when you can just have here: 
```
res.status(200).send(mock.data);
return;
```

**Comment by adamgyulavari** [comment]
@Jeffzholy also the name of the pull request should be also contain the feature summary as it is going to be used as a commit message when squashed into master

**Comment by Jeffzholy** [comment]
resolve conflicts

**Pull request merged**